### PR TITLE
Fix some shellcheck warnings/errors in cluster/gce/util.sh

### PR DESCRIPTION
**Pull request by joakimr-axis**
**What type of PR is this?**
/kind cleanup

**What this PR does / why we need it**:
Fixes shellcheck failures in cluster/gce/util.sh

**Which issue(s) this PR fixes**:
ref: #72956 (part of #90405)
This patch fixes the use of arrays.

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

### CODE REVIEW

**Comment by phenixblue** [comment]
Has it been discussed how we want to handle things with testing for these scripts that have shellcheck fixes broken up into multiple PR's?

**Comment by joakimr-axis** [comment]
> Has it been discussed how we want to handle things with testing for these scripts that have shellcheck fixes broken up into multiple PR's?

The script is run in the CI step, so do you mean checking if the amount of remaining shellcheck issues has indeed been reduced?

**Comment by phenixblue** [comment]
Assuming the plan is to wait until all PR's for this file are merged, removing the script from `hack/.shellcheck_failures` on the final PR. What the order is if one PR needs to be merged before the other. Trying to see if there's an order of priority for getting these reviewed.

**Comment by phenixblue** [comment]
I see most of the PR's from your comment [here](https://github.com/kubernetes/kubernetes/pull/90405#issuecomment-686323844)  have been merged, but from a quick glance I didn't see a PR that actually removed the script from `hack/.shellcheck_failures`.

**Comment by joakimr-axis** [comment]
> I see most of the PR's from your comment [here](https://github.com/kubernetes/kubernetes/pull/90405#issuecomment-686323844) have been merged, but from a quick glance I didn't see a PR that actually removed the script from `hack/.shellcheck_failures`.

Please see the referenced PR #90405, in which the line `./cluster/gce/util.sh` is removed from `hack/.shellcheck_failures`:
 https://github.com/kubernetes/kubernetes/pull/90405/files

**Comment by joakimr-axis** [comment]
> What the order is if one PR needs to be merged before the other.

The sub-PRs have been made to be independent of each other, and the merge order of them should not matter.

**Comment by phenixblue** [comment]
> > I see most of the PR's from your comment [here](https://github.com/kubernetes/kubernetes/pull/90405#issuecomment-686323844) have been merged, but from a quick glance I didn't see a PR that actually removed the script from `hack/.shellcheck_failures`.
> 
> Please see the referenced PR #90405, in which the line `./cluster/gce/util.sh` is removed from `hack/.shellcheck_failures`:
> https://github.com/kubernetes/kubernetes/pull/90405/files

Ok, got it. It wasn't obvious that the original PR was still in play here. That clears things up.

**Comment by joakimr-axis** [comment]
> Ok, got it. It wasn't obvious that the original PR was still in play here. That clears things up.

As some of the sub-PRs have now been merged, the size of the original PR has been reduced and is currently L. With all the sub PRs merged, it should then contain only a small amount of misc leftovers that were not to be categorized into a certain sub PR group.

File: cluster/gce/util.sh
Comment (phenixblue) [inline cluster/gce/util.sh:2054]: Should `kube::util::read-array` be used here for uniformity, or is it purposefully not used here?

File: cluster/gce/util.sh
Comment (phenixblue) [inline cluster/gce/util.sh:2060]: Same here

File: cluster/gce/util.sh
Comment (phenixblue) [inline cluster/gce/util.sh:2072]: Ditto

File: cluster/gce/util.sh
Comment (phenixblue) [inline cluster/gce/util.sh:2102]: Ditto

**Review by phenixblue — COMMENTED** [review]
Just the one main question on the `kube::util::read-array` usage consistency.

File: cluster/gce/util.sh
Comment (joakimr-axis) [reply cluster/gce/util.sh:2054]: `kube::util::read-array` reads line by line and adds it to the array, while here it is a single line that is added. I consider these two separate use cases where `kube::util::read-array` is the best solution for the case for line-by-line, being Bash 3 backwards compatible and all, while this way of writing is the easiest to use/follow/maintain for the single line case.

File: cluster/gce/util.sh
Comment (spiffxp) [inline cluster/gce/util.sh:1597]: I'm not sure this is the same behavior? My read is that `read-array` assumes SERVICE_CLUSTER_IP_RANGE is multi-line, where the previous method allows for space-separated terms

File: cluster/gce/util.sh
Comment (spiffxp) [inline cluster/gce/util.sh:2065]: https://google.github.io/styleguide/shellguide.html#quoting
```suggestion
  if [[ -n "${NODE_LOCAL_SSDS_EXT:-}" ]]; then
```

File: cluster/gce/util.sh
Comment (spiffxp) [inline cluster/gce/util.sh:3696]: why?

**Review by spiffxp — CHANGES_REQUESTED** [review]
Most of these look good, but there are a few I'm uncertain about

File: cluster/gce/util.sh
Comment (joakimr-axis) [reply cluster/gce/util.sh:1597]: Thanks, as you probably noticed I am trying to smoke out what is the problematic change in the equation.
It sure looks like you are right. I will investigate.

File: cluster/gce/util.sh
Comment (joakimr-axis) [reply cluster/gce/util.sh:3696]: > why?

https://github.com/koalaman/shellcheck/wiki/SC2199

I believe what we want to achieve here is the

`Alternatively, if you do want to concatenate all the values in the array into a single string for your test, use "$*" or "${array[*]}" to make this explicit.
`

usecase they describe.

File: cluster/gce/util.sh
Comment (joakimr-axis) [reply cluster/gce/util.sh]: Bullseye, @spiffxp! I did an experiment and left this change out and now all tests pass nicely.
I will look into an updated (=working) solution tomorrow; it is getting late in this neck of the woods.

**Comment by joakimr-axis** [comment]
> Most of these look good, but there are a few I'm uncertain about

With the latest updates, things would be ok now, right?

**Review by spiffxp — APPROVED** [review]
/approve
/lgtm

**Pull request merged**
